src/api/APIFactory.js

```diff
@@ -273,6 +273,17 @@ class APIFactory {
         }
     }
 
+    /**
+     *
+     * @param {Token} token - token used to make requests
+     */
+    setToken(token: Token): void {
+        if (token) {
```

Comment:
minor: Small method so it doesn't matter much, but thoughts on early return like `if (!token) { return; }`?